src/main/java/net/minecraftforge/common/ForgeConfigSpec.java

```diff
@@ -741,6 +838,30 @@ public boolean test(Object t)
             return result;
         }
 
+        public Component getErrorMessage(Object t) {
+            if (isNumber(t))
+            {
+                Number n = (Number) t;
+                boolean result = ((Number)min).doubleValue() <= n.doubleValue() && n.doubleValue() <= ((Number)max).doubleValue();
+                if(!result)
+                {
+                    return Component.translatable("forge.configgui.error.ranged.notInBounds", n.doubleValue(), ((Number)min).doubleValue(), ((Number)max).doubleValue());
```

Comment:
I wonder if this should be just using the string representation of the three numbers rather than displaying the double value as it feels odd to display say `1.0` as a value when it only accepts integers. And then especially if there is a config option for large numbers (longs) where there may not be an exact double representation for it... the error message may confuse people who are like: "that isn't the number I entered"